### ISSUE TITLE
Certain IPv6 Endpoints fail to connect

### ISSUE THREAD

**Issue by caiman-09** [peacey/split-vpn#130]
Hey @peacey 

I've got a really weird issue that I've managed to reproduce a little, but have no idea on how to fix. I think there may be a bug somewhere I can't really find.

I think that certain IPv6 addresses (for some weird reason) can't be used as endpoints.

To try and troubleshoot, I created 10 VMs with my cloud provider and used my server-side WireGuard snapshot to create 10 identical VMs. 

Cycling through these 10 VMs, only 4 worked, and those 4 seemed to have a different IPv6 prefix.

Using the IPv4 address as the endpoint works perfectly on all 10, but I want to phase out my use of IPv4

I initially thought this was some sort of BGP routing issue with my ISP, but I can connect to all 10 VMs when using a local WireGuard client on my PC.

So basically, whether or not I can use this depends on luck and whether I get issued a lucky v6 address from my cloud provider.

When it doesn't work, it just hangs on
 **transfer: 0 B received, 2.60 KiB sent**

Is there anything you can think of which may be causing this?

**Comment by peacey** [comment]
Hi @caiman-09, that's a very weird issue I have not encountered. Can you give me an example of an IPv6 prefix that doesn't work?

**Comment by peacey** [comment]
Closing this since I helped you out on Reddit. The problem was OP was using a global IPv6 address for the internal wireguard address, and the UDM doesn't have a global IPv6 assigned on the WAN interface (only on LAN interfaces). OP also was using an IPv6 wireguard endpoint, and this meant the UDM was using the internal wireguard address as a source address from the WAN interface, instead of the proper global IPv6 from the brX interfaces when trying to initiate the handshake with the endpoint.

The two solutions were: 

1. Easiest solution: Don't use a publicy-routable IPv6 address for the wireguard internal address. Instead use a private one that starts with fd or fc.
2. Harder solution: Add a global IPv6 to the ethX/pppX WAN interface based on the IPv6 prefix assigned to brX. This works but requires more effort because you have to watch the IPv6 incase it needs to be changed or is removed.
